### PR TITLE
H-08

### PR DESCRIPTION
**Pull request by 0xMithrandir**
Fix for Large Orders Can Exceed minimumCredit finding H-04

### CODE REVIEW

File: markets/bfp-market/contracts/storage/PerpMarket.sol [hunk 1]
```diff
@@ -133,7 +133,7 @@ library PerpMarket {
         int256 fundingDelta = newPosition.entryFundingAccrued.mulDecimal(sizeDelta);
         int256 notionalDelta = newPosition.entryPrice.toInt().mulDecimal(sizeDelta);
         int256 totalPositionPnl = oldPosition.getPricePnl(newPosition.entryPrice) +
-            oldPosition.getAccruedFunding(self, newPosition.entryPrice);
+                            oldPosition.getAccruedFunding(self, newPosition.entryPrice);
```
Comment (0xjocke) [inline markets/bfp-market/contracts/storage/PerpMarket.sol:136]: Format issue

Comment (0xMithrandir) [reply markets/bfp-market/contracts/storage/PerpMarket.sol]: Updated

File: markets/bfp-market/test/integration/modules/OrderModule.test.ts [hunk 2]
```diff
@@ -104,6 +104,36 @@ describe('OrderModule', () => {
       );
     });
 
+    it('should revert on a too large order', async () => {
```
Comment (davidvuong) [inline markets/bfp-market/test/integration/modules/OrderModule.test.ts:107]: You can make this test a little bit less brittle by calculating the actual min credit then, similar to other tests, generate deposit values that exceed the min credit. Also consider renaming this test to mention that this is testing for min credit

File: markets/bfp-market/contracts/storage/Position.sol [hunk 3]
```diff
@@ -295,7 +302,13 @@ library Position {
             );
 
             // Check the minimum credit requirements are still met.
-            validateMinimumCredit(market, params.oraclePrice, marketConfig, addresses);
+            validateMinimumCredit(
```
Comment (davidvuong) [inline markets/bfp-market/contracts/storage/Position.sol:305]: You should also test that the same revert occurs on settlement if min credit checks had passed at the time of commitment.

File: markets/bfp-market/contracts/storage/PerpMarket.sol [hunk 4]
```diff
@@ -185,6 +185,21 @@ library PerpMarket {
             self.depositedCollateral[addresses.sUsd];
     }
 
+    /// @dev Returns the market's required minimum backing credit in USD.
+    function getMinimumCreditWithTradeSize(
+        PerpMarket.Data storage self,
+        PerpMarketConfiguration.Data storage marketConfig,
+        uint256 price,
+        AddressRegistry.Data memory addresses,
```
Comment (davidvuong) [inline markets/bfp-market/contracts/storage/PerpMarket.sol:193]: Minor: You'll notice that `addresses` is also the last arg in all internal functions. Can we keep this consistent with that please.

Comment (0xMithrandir) [reply markets/bfp-market/contracts/storage/PerpMarket.sol:193]: Yes, of course, I haven't noticed that but I'll update it.

File: markets/bfp-market/contracts/modules/PerpMarketFactoryModule.sol [hunk 5]
```diff
@@ -172,6 +172,25 @@ contract PerpMarketFactoryModule is IPerpMarketFactoryModule {
             );
     }
 
+    function minimumCreditWithTradeSize(uint128 marketId, int128 sizeDelta) external view returns (uint256) {
```
Comment (davidvuong) [inline markets/bfp-market/contracts/modules/PerpMarketFactoryModule.sol:175]: Add missing docs above

**Review by davidvuong — APPROVED** [review]
Last comment on docs in `minimumCreditWithTradeSize` but should be ok to merge.

**Pull request merged**